### PR TITLE
Updated logging capabilities

### LINKED ISSUE #394: Gusto logger should produce a log file

### PR DESCRIPTION
**Pull request by JDBetteridge**
Fixes #394

### CODE REVIEW

File: gusto/timeloop.py [hunk 1]
```diff
@@ -3,14 +3,14 @@
 from abc import ABCMeta, abstractmethod, abstractproperty
 from firedrake import Function, Projector, Constant, split
 from pyop2.profiling import timed_stage
-from gusto.configuration import logger
 from gusto.equations import PrognosticEquationSet
-from gusto.forcing import Forcing
+from gusto.fields import TimeLevelFields, StateFields
```
Comment (tommbendall) [inline gusto/timeloop.py:7]: Thanks for alphabetising this!

File: gusto/configuration.py [hunk 2]
```diff
@@ -87,12 +68,15 @@ def __setattr__(self, name, value):
             AttributeError: if the :class:`Configuration` object does not have
                 this attribute pre-defined.
         """
+        if name == 'log_level':
+            logger.warning("Set log level with environemnt variables, see `logging.py` for details")
```
Comment (tommbendall) [inline gusto/configuration.py:72]: Typo here

Comment (tommbendall) [inline gusto/configuration.py:72]:
```suggestion
            logger.warning("Set log level with environment variables, see `logging.py` for details")
```

**Comment by JDBetteridge** [comment]
There's more to come, I think I can get KSP residuals to be logged. I think what we need is a custom KSP monitor, like I have written [here](https://github.com/JDBetteridge/gusto_linesmoother/blob/master/monitor.py) (but _much_ simpler), which logs the exact same information that is output by `ksp_monitor_true_residual` to the Python logger and then turn off the PETSc internal logger to avoid duplication.

**Comment by JDBetteridge** [comment]
I think this is ready for another review :slightly_smiling_face:

**Review by tommbendall — APPROVED** [review]
This is really good. I can confirm that I have run an example in serial and in parallel on my machine and that the log files appear in the results directory for that example.

Thanks very much!

**Pull request merged**